### PR TITLE
RTreeDisk - disk_backed

### PR DESCRIPTION
**Pull request by sergiupocol**
src/include/rtreedisk/*  contains the implementation for RTreeDisk and it's corresponding node class.

src/tests/testRTreeDisk.cpp contains the tests the cover the main functions in the Index.h interface along with findLeaf and findLeaf on Disk

### CODE REVIEW

File: src/include/rtreedisk/node.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        tree_node_handle findLeaf(Point givenPoint);
+        void moveData(unsigned fromIndex, std::vector<Point> &toData);
+        void moveChild(unsigned fromIndex, std::vector<Rectangle> &toRectangles, std::vector<tree_node_handle> &toChildren);
+        void moveData(std::vector<Point> &fromData);
+        void moveChildren(std::vector<tree_node_handle> &fromChildren, std::vector<Rectangle> &fromBoxes);
+        void addEntryToNode(Rectangle box, tree_node_handle child)
+        {
+            using NodeType = Node<min_branch_factor, max_branch_factor>;
+            tree_node_allocator *allocator = get_node_allocator(treeRef);
+            allocator->get_tree_node<NodeType>(child)->parent = self_handle_;
```
[the remark this answers is not in the document]
Comment (sergiupocol) [reply src/include/rtreedisk/node.h:90]: Yes, it's usage is strictly in testRTreeDisk and in that context I believe addEntry is only called on the parent.

File: src/tests/testRTreeDisk.cpp [hunk 2]
```diff
@@ -7,6 +7,15 @@
 
 using NodeType = rtreedisk::Node<3, 6>;
 using TreeType = rtreedisk::RTreeDisk<3, 6>;
+using BranchType = NodeType::Branch;
+
+template <class NE, class B>
+static NE createBranchEntry(const Rectangle
```
[the remark this answers is not in the document]
Comment (bglasber) [reply src/tests/testRTreeDisk.cpp:13]: If you do end up keeping the duplicates and put the _local version here instead, you can try using the following VIM command trick:

:%s/createBranchEntry/createBranchEntry_local/g

**Review by bglasber — CHANGES_REQUESTED** [review]
Just a few final things regarding naming and using std::get<> & references where appropriate.

Just tested with the California dataset (2D, 1.8 million points) and it all loads and validates successfully. Great work! 

I will break off a task from the NIR-Tree allocator optimizations and talk to you mid-week about how we get can started on that.

File: src/include/rtreedisk/node.tcc [hunk 3]
```diff
@@ -84,8 +97,8 @@ void Node<min_branch_factor, max_branch_factor>::removeChild(tree_node_handle ch
 {
     for (unsigned i = 0; i < cur_offset_; i++)
     {
-
-        if (children[i] == child)
+        Branch b = std::get<Branch>( entries.at(i) );
```
[the remark this answers is not in the document]
Comment (sergiupocol) [reply src/include/rtreedisk/node.tcc:102]: Thanks! I think all instances of Branch b = ... are not modifying b but just reading so all can be converted to references

**Review by bglasber — APPROVED** [review]

**Pull request merged**